pkg/worker/client.go

```diff
@@ -70,9 +87,19 @@ func (c ScrapeAndDumpClient) Poll(ctx context.Context, errC chan error) {
 				return
 			default:
 			}
-			err := c.scrapeAndDump(ctx)
-			if err != nil {
-				c.logger.Error(err.Error())
+			var err error
+			if c.cb != nil {
+				err = c.cb.Run(func() error { return c.scrapeAndDump(ctx) })
+				if err != nil && err == circuitbreaker.ErrSystemFailure {
+					errC <- err
+					return
+				}
+			} else {
+				err := c.scrapeAndDump(ctx)
+				if err != nil {
```

Comment:
So is the default behavior (when no circuit breaker is in use) that all errors are fatal? I would lean towards the opposite but I'd like to hear the rationale